keps/sig-api-machinery/1281-network-proxy/README.md

```diff
@@ -403,6 +403,23 @@ Beta:
   are spending dialing out.
 - Ensure we have metrics on each existing egress use case.
 
+GA:
+
+- TBD
+- Protocol simplification: [#462](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/462)
+- Post-dial instrumentation: [#410](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/410)
+- Overload protection.
+  * Proxy client use.
+  * Agent connections.
+- Version skew test coverage.
+- Benchmarks and stress tests: [#366](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/366)
+- Efficiency improvements? [#180](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/180), [#236](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/236)
+- Dynamic Server count.
+  * Complete Server half: [#273](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/273)
```

Comment:
I (on behalf of the hypershift project) would love to see the ability to control at a finer level what traffic goes through this component in the egress config as well. 

The environment to think about is a split control plane/data plane environment where a set of webhooks/apiservices are in a colocated network with the kube-apiserver (and therefore have direct IP connectivity to those webhooks/apiservices. However: there are a subset of "in cluster/data plane" apiservices/webhooks that will need to proxy through konnectivity in order to hit the appropriate endpoint (in cluster services like metrics apiservice, in cluster webhooks like istio, etc). 

Right now it's an all or nothing approach for specifying if webhook and apiservice traffic goes through konnectivity or direct. To work around this: a user has to deploy agents in both the "control plane network" that advertise the set of ips for the webhook traffic and then traffic has to go kube-apiserver -> konnectivity server -> konnectivity agent -> webhook backend even though it can optimally go kube-apiserver -> webhook backend

Current workaround is the konnectivity agents in the control plane network advertises the "subset" of ips that exist in the control plane: and then the suite of konnectivity agents deployed in the cluster advertise a "default route" for all other ips.

Reply:
I can document this in an issue if one doesn't exist but wanted to see if this makes sense to everyone here before doing so. This likely involves updated the "Egress Configuration" spec

Reply:
@relyt0925 @jkh52 
 
This is a valid scenario and I would love to have that finer control as well.

We (AKS) have had a scenarios exactly like you've described. We too have drafted a similar workaround of having "konnectivity-agents" in the control plane that would help with webhook traffic.

I am +1 on this feature to be added. I'll add this to the agenda to be discussed in the apiserver-network-proxy sync meeting as I think it makes sense for GA.

Reply:
@relyt0925 It would be great if you could please create an issue in https://github.com/kubernetes-sigs/apiserver-network-proxy/issues

Reply:
@relyt0925 please do open an issue, it looks worth discussing.

Reply:
^you are correct (and also correct that it's not for extension servers)

Reply:
Took a initial stab at documenting:
https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/496

Reply:
^ @ipochi happy to join the sync meeting as well!

Reply:
Thanks for the documentation, it helps explain the scenario.  The feature itself may have merit, but such capability would like on the kube-apiserver side and be a significant shift to the current feature both in terms of configuration and implementation.  Rather than make that shift and couple it this current effort, starting a new KEP for expansion of the featureset looks more appropriate.